scheduler/task.go

```diff
@@ -295,8 +295,11 @@ func (t *task) stream() {
 			}
 			select {
 			case <-t.killChan:
+				t.Lock()
 				t.state = core.TaskStopped
-				break
+				t.Unlock()
+				done = true
```

Comment:
After removing g.killed() in control/plugin/client/grpc.go, where are we closing the channel to be able to receive "closing of killChan" here?